src/solver/impl/include/sem_solver_impl.h

```diff
@@ -190,11 +154,14 @@ template <int ORDER, typename INTEGRAL_TYPE, typename MESH_TYPE>
 void SEMsolver<ORDER, INTEGRAL_TYPE, MESH_TYPE>::updatePressureField(
     float dt, int i1, int i2, const ARRAY_REAL_VIEW &pnGlobal)
 {
+
+  bool const isModelOnNodes = m_mesh.isModelOnNodes();
   float const dt2 = dt * dt;
   LOOPHEAD(m_mesh.getNumberOfNodes(), I)
   {
-    pnGlobal(I, i1) = 2 * pnGlobal(I, i2) - pnGlobal(I, i1) -
-                      dt2 * yGlobal[I] / massMatrixGlobal[I];
+    real_t const density = isModelOnNodes ? m_mesh.getModelRhoOnNodes(I) : 1.0f;
```

Comment:
I discuss this morning with Stefano about that, it is a little bit more tricky then I thought at the beginning, we can't put it here but we also can't do it the way I did in the first place I need to modify it.